sktime/transformations/series/bootstrapping.py

```diff
@@ -0,0 +1,395 @@
+        )
+        self.box_cox_transformer_.fit(X)
+
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _transform must support all types in it
+            Data to be transformed
+        y : Series or Panel of mtype y_inner_mtype, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        """
+        X_index = X.index
+
+        X_transformed = self.box_cox_transformer_.transform(X)
+
+        # fit STL on X_transformed series and extract trend, seasonal and residuals
+        stl = _STL(
+            X_transformed,
+            period=self.sp,
+            seasonal=self.seasonal,
+            trend=self.trend,
+            low_pass=self.low_pass,
+            seasonal_deg=self.seasonal_deg,
+            trend_deg=self.trend_deg,
+            low_pass_deg=self.low_pass_deg,
+            robust=self.robust,
+            seasonal_jump=self.seasonal_jump,
+            trend_jump=self.trend_jump,
+            low_pass_jump=self.low_pass_jump,
+        ).fit(inner_iter=self.inner_iter, outer_iter=self.outer_iter)
+        seasonal = pd.Series(stl.seasonal, index=X_index)
+        resid = pd.Series(stl.resid, index=X_index)
+        trend = pd.Series(stl.trend, index=X_index)
+
+        # time series id prefix
+        prefix = self.series_name + "_" if self.series_name is not None else ""
+
+        # initialize the dataframe that will store the bootstrapped series
+        if self.return_actual:
+            df = pd.DataFrame(
+                X.values,
+                index=pd.MultiIndex.from_product(
+                    iterables=[[f"{prefix}actual"], X_index],
+                    names=["series_id", "time_index"],
+                ),
+            )
+        else:
+            df = pd.DataFrame()
+
+        # create multiple series
+        for i in range(self.number_of_new_series):
+            new_series = self.box_cox_transformer_.inverse_transform(
+                self.moving_block_bootstrap(
+                    ts=resid,
+                    block_length=self.block_length_,
+                    replacement=self.sampling_replacement,
+                )
+                + seasonal
+                + trend
+            )
+
+            new_series_id = f"{prefix}synthetic_{i}"
+            new_df_index = pd.MultiIndex.from_product(
+                iterables=[[new_series_id], new_series.index],
+                names=["series_id", "time_index"],
+            )
+
+            df = df.append(pd.DataFrame(data=new_series.values, index=new_df_index))
+
+        return df
+
+    @staticmethod
+    def moving_block_bootstrap(
```

Comment:
```suggestion
    def _moving_block_bootstrap(
```

Reply:
keeping it static, but making it private